### PR TITLE
[bitnami/zookeeper] Rename metrics port and expose scheme, and tlsConfig for Istio compatibility

### PR DESCRIPTION
**Pull request by cpepper96**
### Description of the change

Hi folks, this PR is to rename the `port` name for the metrics Service and to expose the `scheme` and `tlsConfig` fields for the ServiceMonitor. 

### Benefits

This will enable Prometheus metrics to be configured properly when deploying ZooKeeper to a cluster configured with Istio. The metrics service port name is `tcp-metrics` causing Istio to [automatically select](https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection) the TCP protocol for this service which causes an error. By changing this to `http-metrics` Istio will correctly select the `http` protocol.

Similarly for the ServiceMonitor object, exposing the `scheme` and `tlsConfig` will enable Prometheus metrics to be integrated with Istio. 

### Possible drawbacks

I made the `scheme`, and `tlsConfig` values optional and empty by default to ensure backwards compatibility.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/zookeeper/templates/metrics-svc.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ spec:
     - name: tcp-metrics
       port: {{ .Values.metrics.service.port }}
       targetPort: metrics
+      {{- if .Values.metrics.service.appProtocol }}
+      appProtocol: {{ .Values.metrics.service.appProtocol }}
+      {{- end }}
```
Comment (juan131) [inline bitnami/zookeeper/templates/metrics-svc.yaml:26]: Can't we simply rename the port name to `http-metrics`?

File: bitnami/zookeeper/Chart.yaml [hunk 2]
```diff
@@ -28,4 +28,4 @@ maintainers:
 name: zookeeper
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/zookeeper
-version: 13.4.14
+version: 13.4.15
```
Comment (juan131) [inline bitnami/zookeeper/Chart.yaml:31]: Please bump minor version instead given you're adding new chart parameters

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for this contribution! Please take a look at my comments when you have a chance.

File: bitnami/zookeeper/templates/metrics-svc.yaml (hunk 1 above)
Comment (cpepper96) [reply bitnami/zookeeper/templates/metrics-svc.yaml:26]: Yep! That works as well, we would just need to update the port name for the ServiceMonitor to match. 

I considered that approach but I thought that might be too big of a lift and potentially break some deployments. What are your thoughts?

Comment (juan131) [reply bitnami/zookeeper/templates/metrics-svc.yaml:26]: What deployments do you refer to? I don't think other than metrics collector such as Prometheus would be interested on connecting to Zookeeper metrics service port. And, given we'll be adapting the ServiceMonitor in the same PR, I don't expect breaking changes.
In any case, you can always test your changes by running a `helm upgrade` after applying your changes and see if you find any errors.

Comment (cpepper96) [reply bitnami/zookeeper/templates/metrics-svc.yaml:26]: I was thinking if someone was deploying/managing the ServiceMonitor outside of the helm chart they might not pick up on that port name change. That would probably be a rare case though.

I'd prefer to change the port name to solve this problem anyways so if it's good with you then I can implement that!

Comment (juan131) [reply bitnami/zookeeper/templates/metrics-svc.yaml:26]: Yes please, change the port name so the `metrics.service.appProtocol` parameter isn't required.

Comment (cpepper96) [reply bitnami/zookeeper/templates/metrics-svc.yaml]: Made those updates and updated the PR title/description to reflect that. Thanks for the help!

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**